### DISCUSSION TITLE
Fabm/eco model in a flooding and drying environment

### DISCUSSION THREAD

**Discussion by jpein1** [fabm-model/fabm#36] [Q&A]
Hi All,

Using Fabm in a flooding-and-drying/wetting-and-drying environment, I noticed that particulate stuff accumulates on tidal flats. This is somewhat normal, however it shouldn’t be that when a cell is dry, the ecologic model called by Fabm is completely at rest such that sinks that do not require a water cover are switched off. 
In fact, I am looking for a way to call the “do bottom” only, and only account for the sinks like denitrification and burial of particulate matter (process accounted for by the eco model). 
Is there a preferred way to tackle this with Fabm? One idea I had in mind is to make the “do bottom” subroutine public to be able to call it separately from the pelagic (main) eco model part. Would this be feasible or do you know a better work around? 

Best,
Johannes

**Comment by jornbr** [comment]
Hi Johannes,

One way to achieve this would be to have the host call all FABM routines (`get_interior_sources`, `get_bottom_sources`, `get_surface_sources`, etc.) for both wet and dry points, and then in the returned source arrays, to ignore (e.g., zero) any terms associated with the pelagic in dry points. This offers the best guarantee that all benthic source terms are computed and that all model diagnostics will be correct in dry points  - and without having to hack into either FABM or the biogeochemical model. The additional overhead of pelagic routines being called in dry points is likely neglegible, unless a very large part of your domain is dry over long periods.

However, it seems to me that it will depend heavily on the biogeochemical model whether this produces the right effect. If you allow bottom sources terms (i.e. changes in benthic state variables), you will likely also have to allow benthic-pelagic fluxes. If not, mass conservation would be violated in many BGC models. These benthic-pelagic fluxes will modify the pelagic state. If surface and pelagic dynamics are then switched off, the pelagic state variables could drift heavily while the cell is dry, and in some cases, that drift would be unrealistic. For instance, for dissolved gases, air-sea exchange would in reality stabilize the concentrations.

A more refined but also more involved solution could be to make the BGC model aware of cells being dry, by introducing a dependency that takes value 0 for dry, 1 for wet (for instance). The host could then pass that field, leaving it to individual BGC modules to decide which processes they leave active.

All in all, this seems to require some experiments, potentially with surprising/unexpected outcomes. It'd be interesting to hear how it works out for you.

Cheers,

Jorn
